examples/ex06_realtime_move.cpp

```diff
@@ -19,20 +19,27 @@ class JpCircle : public systems::SingleIO<double, typename units::JointPositions
 
 public:
 	JpCircle(jp_type startPos, double amplitude, double omega, const std::string& sysName = "JpCircle") :
-		systems::SingleIO<double, jp_type>(sysName), jp(startPos), j3_0(jp[2]), j4_0(jp[3]), amp(amplitude), omega(omega) {}
+		systems::SingleIO<double, jp_type>(sysName), jp(startPos), jp_0(jp), amp(amplitude), omega(omega) {}
 	virtual ~JpCircle() { this->mandatoryCleanUp(); }
 
 protected:
 	jp_type jp;
-	double j3_0, j4_0;
+	jp_type jp_0;
 	double amp, omega;
 	double theta;
 
 	virtual void operate() {
 		theta = omega * this->input.getValue();
 
-		jp[2] = amp * std::sin(theta) + j3_0;
-		jp[3] = amp * (std::cos(theta) - 1.0) + j4_0;
+    if (DOF>3) {
+      // WAM - Use joints 3 and 4.
+      jp[2] = amp * std::sin(theta) + jp_0[2];
+      jp[3] = amp * (std::cos(theta) - 1.0) + jp_0[3];
+    } else {
+      // Proficio - Use joints 2 and 3.
+      jp[1] = amp * std::sin(theta) + jp_0[1];
+      jp[2] = amp * (std::cos(theta) - 1.0) + jp_0[2];
+    }
 
```

Comment:
Instead of checking the DOF inside the operate which loops continuously, can we perform the check once in the constructor and assign variables to it. Something like this.
JpCircle(...) : ... {
   if (DOF>3) {
      i =2;
      j = 3;
  } else {
     i = 1;
     j =2;
  }
}
int i,j;
.
.
.
void operate() {
.
.
.
jp[i] = ....... jp_0[i];
jp[j] = ....... jp_0[j];